src/logic/calculate.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable  prefer-template  */
+
+
+import Big from 'big';
+
+import operate from './operate';
+import isNumber from './isNumber';
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:String      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
+ */
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
+    return {
+      total: null,
+      next: null,
+      operation: null,
+    };
+  }
+
+  if (isNumber(buttonName)) {
+    if (buttonName === '0' && obj.next === '0') {
+      return {};
+    }
+    // If there is an operation, update next
+    if (obj.operation) {
+      if (obj.next) {
+        return { next: obj.next + buttonName };
+      }
+      return { next: buttonName };
+    }
+    // If there is no operation, update next and clear the value
+    if (obj.next) {
+      const next = obj.next === '0' ? buttonName : obj.next + buttonName;
+      return {
+        next,
+        total: null,
+      };
+    }
+    return {
+      next: buttonName,
+      total: null,
+    };
+  }
+
+  if (buttonName === '%') {
+    if (obj.operation && obj.next) {
+      const result = operate(obj.total, obj.next, obj.operation);
+      return {
+        total: Big(result)
+          .div(Big('100'))
+          .toString(),
+        next: null,
+        operation: null,
+      };
+    }
+    if (obj.next) {
+      return {
+        next: Big(obj.next)
+          .div(Big('100'))
+          .toString(),
+      };
```

Comment:
**[ OPTIONAL ]** This use of the percentage sign could go into the operate file as it is intended to be used as an operation and not a modifier, as we can see in the last paragraph of point number 3 [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896094/) 
![image](https://user-images.githubusercontent.com/46115725/72094285-b4c11200-32db-11ea-9365-c07939230d09.png)
 I'm marking it as optional as there is no hard requirement for you to do it since it will have to be inferred from the info given.